docs/sdk/api/sdkAuth.md

```diff
@@ -244,3 +251,69 @@ await authClient.introspectToken('invalid_token')
 //   "active":false
 // }
 ```
+
+### Token Provider
+
+Token provider is a special class which watches over `access_token` and refreshes it using `sdkAuth.refreshTokenFlow()` method if it is expired.
+
+#### Constructor argument
+
+1.  `options` _(Object)_: Configuration object
+    - `sdkAuth` - SdkAuth object initialized with project credentials
+    - `getTokenInfo` - Optional function which will be called for retrieving `access_token` if the `tokenInfo` or `refresh_token` were not provided.
+    - `onTokenInfoChanged` - Optional function which is being called when the tokenInfo gets changed (manually or by refresh process)
+    - `onTokenInfoRefreshed` - Optional function which is being called when the tokenInfo gets refreshed
+2.  `tokenInfo` _(Object)_: Optional parameter containing token information loaded from one of auth flows (`{ access_token, refresh_token, expires_at }`)
+
+#### Usage example
+
+Minimal example:
+
+```js
+import SdkAuth, { TokenProvider } from '@commercetools/sdk-auth'
+
+// initiate TokenProvider
+const tokenProvider = new TokenProvider({
+  sdkAuth: new SdkAuth({
+    // .. init SdkAuth
+  }),
+  getTokenInfo: sdkAuth => sdkAuth.clientCredentialsFlow(),
+})
+
+const accessToken = await tokenProvider.getToken()
```

Comment:
I think there are some inconsistencies on the name of the `getTokenInfo` (as the method that allows me (as dev) to get the token from the tokenProvider), and `fetchTokenInfo` (as the method that I pass as parameter that allows the provider to keep the token alive). I've seen `getTokenInfo` being used in both situations.

Reply:
Hi @lauraluiz  .. please check my latest commit 166e4cb